packages/components/src/DataList/components/DataListLayoutInternal/DataListItems.tsx

```diff
@@ -19,41 +23,83 @@ export function DataListItems<T extends DataListObject>({
   mediaMatches,
   data,
 }: DataListItemsProps<T>) {
+  const { itemActionComponent } = useDataListContext();
   const elementData = generateListItemElements(data);
```

Comment:
this is a relatively expensive calculation with a `reduce` nested in a `map` where the `data` we're mapping can keep growing as we load more

due to the mouse enter and mouse leave setting state inside this component, this will get recalculated a lot of times moving your cursor down the list

let's memo-ize this value with `data` as the only dependency to make sure we're only ever recalculating it when we need to!

Reply:
first, thanks for making that change. this is much better now!

second, I do wonder if there's more optimization for us to do here. even with this, we're doing thousands and thousands of iterations as we get to the end of the list with only ~80 items

![image](https://github.com/GetJobber/atlantis/assets/11843215/1bb84b32-9342-470e-9a2a-223fad0740ca)

I feel like we should be able to cache or memo-ize even further here, because for each new set of data added we have already gone through the process of generating the elements for everything before. I don't think there's any value in generating them again because there's nothing that would have changed for them.

ideally we leverage the work we've already done, and then just do what's necessary for that new data

that said, I know that's not the point of this PR so I'm not going to block on that but it's something we should decide if we feel is tech debt/worth following up on

Reply:
yeah, i'm not totally fond of how we did the generation of the items. I think that's a good optimization to do. I'll add a tix to get it looked at.

With that said, it is mostly regenerating the code behind it. Luckily for us, the `key` is preventing everything from being rendered in the DOM.